src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonNamingPolicy.cs

```diff
@@ -20,6 +20,15 @@ public abstract class JsonNamingPolicy
 
         internal static JsonNamingPolicy Default { get; } = new JsonDefaultNamingPolicy();
 
+        /// <summary>
+        /// Returns the naming policy for snake-casing.
+        /// </summary>
+        public static JsonNamingPolicy SnakeCase { get; } = new JsonSnakeCaseNamingPolicy();
+
+        // KebabCase was added together with SnakeCase because sharing implementation.
```

Comment:
Generally we don't add code in case it might be exposed in future. We would add the implementation when the API is approved.

Reply:
Doesn't [this](https://github.com/dotnet/runtime/issues/782) appear as approved?